internal/store/hpa.go

```diff
@@ -76,6 +78,91 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_hpa_spec_target_metric",
+			Type: metric.Gauge,
+			Help: "The metric specifications used by this autoscaler when calculating the desired replica count.",
+			GenerateFunc: wrapHPAFunc(func(a *autoscaling.HorizontalPodAutoscaler) *metric.Family {
+				targets := make([]*metric.Metric, 0, len(a.Spec.Metrics))
+				for _, m := range a.Spec.Metrics {
+					switch m.Type {
+					case autoscaling.ObjectMetricSourceType:
+						targetValue, ok := m.Object.TargetValue.AsInt64()
+						if ok {
+							targets = append(targets, &metric.Metric{
+								LabelKeys:   targetMetricLabels,
+								LabelValues: []string{m.Object.MetricName, "value"},
+								Value:       float64(targetValue),
+							})
+						}
+						if m.Object.AverageValue != nil {
+							averageValue, ok := m.Object.AverageValue.AsInt64()
+							if ok {
+								targets = append(targets, &metric.Metric{
+									LabelKeys:   targetMetricLabels,
+									LabelValues: []string{m.Object.MetricName, "average"},
+									Value:       float64(averageValue),
+								})
+							}
+						}
+					case autoscaling.PodsMetricSourceType:
```

Comment:
We can condense this switch-case block further. Let's just assign the label values and metric values based on the hpa target metric type. After that, we can have just one instance of the target metric creation and the append to the metrics slice.